instrumentation/logger/lib/opentelemetry/instrumentation/logger/patches/logger.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Logger
+      module Patches
+        # Instrumention for methods from Ruby's Logger class
+        module Logger
+          # TODO: Make sure OTel logs aren't instrumented
+          # TODO: How to pass attributes?
+          def format_message(severity, datetime, progname, msg)
+            formatted_message = super(severity, datetime, progname, msg)
+            return formatted_message if skip_instrumenting?
+
+            logger_provider.logger(
+              name: @config[:name],
+              version: @config[:version]
+            ).on_emit(
+              severity_text: severity,
+              severity_number: severity_number(severity),
+              timestamp: datetime,
+              body: msg, # New Relic uses formatted_message here. This also helps us with not recording progname, because it is included in the formatted message by default. Which seems more appropriate?
+              context: OpenTelemetry::Context.Current
+            )
+            formatted_message
+          end
+
+          option :name, default: OpenTelemetry::Instrumentation::Logger::NAME, validate: :string
+          option :version, default: OpenTelemetry::Instrumentation::Logger::VERSION, validate: :string
```

Comment:
I think these lines need to be in [logger/instrumentation.rb](https://github.com/kaylareopelle/opentelemetry-ruby-contrib/blob/logger-instrumentation/instrumentation/logger/lib/opentelemetry/instrumentation/logger/instrumentation.rb). [Here's an example for postgres instrumentation.](https://github.com/kaylareopelle/opentelemetry-ruby-contrib/blob/logger-instrumentation/instrumentation/pg/lib/opentelemetry/instrumentation/pg/instrumentation.rb#L27-L29)